### DISCUSSION TITLE
Improvement: Better error message for unregistered base class

### DISCUSSION THREAD

**Discussion by silabs-Vladimir** [wjakob/nanobind#346] [General]
Ref this issue: https://github.com/wjakob/nanobind/issues/344

I find that the error message when attempting to call a subclass function without having registered the base class:

```
E       TypeError: get_temporary_buffer_size(): incompatible function arguments. The following argument types are supported:
E           1. get_temporary_buffer_size(self, arg: ndarray[dtype=uint8], /) -> int
E       
E       Invoked with types: transforms_ext.CastTransform_uint8_t_int8_t, ndarray
```

is very misleading and should be more specific if possible.

**Comment by wjakob** [comment]
I don't see an easy way to improve the error message without imposing a major implementation-level or runtime cost. If you want to give it a try, PRs are welcome.
